### ISSUE TITLE
Reveal.js PDF Export menu item is broken

### ISSUE THREAD

**Issue by awehrfritz** [quarto-dev/quarto-cli#2988]
In reveal.js presentations, the menu item for the PDF Export tool triggers the overview mode rather than a PDF export.

Looking at what I believe is the respective code, it appears to be a simple oversight and probably straightforward to fix:
https://github.com/quarto-dev/quarto-cli/blob/7cc0ced8c9d62a8bc6a3ed8863160f90729b77bf/src/format/reveal/format-reveal-plugin.ts#L438-L442

Weirdly, though, the key binding "e" still works as expected.

**Comment by jjallaire** [comment]
It looks like there is no way for us to programmatically invoke PDF Export mode (except perhaps sending a fake 'e' key through).

**Comment by cderv** [comment]
Could a solution be to PR to upstream [PdfExport](https://github.com/McShelby/reveal-pdfexport) plugin that we use so that the [`togglePdfExport()`](https://github.com/McShelby/reveal-pdfexport/blob/d8d4c6d22f4153419a64cd9d5f46f0c55487f36d/pdfexport.js#L59-L69) is exported and thus can be used in JS like we can do with `RevealChalkboard`  ?  This is the callback function for keybinding 'e'.

**Comment by jjallaire** [comment]
Yes, definitely! We could also fork this plugin and add that feature.

**Comment by awehrfritz** [comment]
Great to see some activity on this issue! As I am not familiar with the Quarto or reveal.js codebase, I cannot really contribute much to the technical solution. 

However, as far as I can tell, Quarto ships already a copy of this plugin:
[src/resources/formats/revealjs/plugins/pdfexport/pdfexport.js](https://github.com/quarto-dev/quarto-cli/blob/60162fa59757b0948e66f59cb369eaba5ade6385/src/resources/formats/revealjs/plugins/pdfexport/pdfexport.js)

Thus, changes to implement this feature could indeed be made easily in Quarto directly (and possibly up-streamed later on).

**Comment by cderv** [comment]
> Thus, changes to implement this feature could indeed be made easily in Quarto directly (and possibly up-streamed later on).

Yes we ship a copy that we update from time to time when we decide to update our dependency version. We could definitely start to make a change there, and then update upstream. 

@jjallaire What do you think ? 

* I can make the change in our plugin file, and having a diff patch file to apply when we update dependency (in [`update-html-dependencies.ts`](https://github.com/quarto-dev/quarto-cli/blob/main/package/src/common/update-html-dependencies.ts)

* Then if we think it is better, I can PR to upstream. 

I'll take the assignment on that

**Comment by jjallaire** [comment]
@cderv Yes I think that would be an excellent approach! Definitely a good thing to do for v1.3

**Comment by mash-graz** [comment]
Even if the menu gets fixed, it's still a very problematic solution, which doesn't work reliable on most browsers. For example on Firefox you will still get garbage resp. wrong page breaks in the printed results after activating this mode. Well, you'll find a warning note about this flaw in the documentation, but it's IMHO still a very disappointing behavior. 

I would therefore prefer a more reliable solution, which provides reproducible correct results in arbitrary browsers.

This can be realized in practice by integrating [decktape](https://github.com/astefanutti/decktape) resp. [puppeteer](https://github.com/puppeteer/puppeteer) or a similar mechanism into the CLI-tool for the document processing and just add an additional download button in the menu or toolbar instead of this present print style entry. That's the way, how for example [slidev](https://github.com/slidevjs/slidev) handles this task in a much more reliable manner.

**Comment by jjallaire** [comment]
Yes, I agree that the current solution is quite awkward. Best would be if PDF output could be fully integrated into `quarto preview` (so you can always be working w/ the PDF version of the slides). We could accomplish this with decktape for sure.

**Comment by jjallaire** [comment]
@cderv If we do complete the menu fix for v1.3 let's open up a new issue to do deck tape in v1.4 or v1.5

**Comment by cderv** [comment]
I have done the fix in #2988 and should be merged soon. Isssue opened for the decktape support feature #4677
